xarray/core/dataset.py

```diff
@@ -6638,8 +6653,23 @@ def pad(
             coord_pad_options = {}
 
         variables = {}
+
+        # standarize pad_width
+        pad_width_standarized = {}  # type: Mapping[Hashable, Tuple[int, int]]
```

Comment:
Needs help for correcting the typing.
Now, `pad_width_standarized` should be `Mapping[Hashable, Tuple[int, int]]` instead of `Mapping[
            Hashable, Union[int, Tuple[Union[int, Iterable], Union[int, Iterable]]]
        ]`
mypy is complaining...